worker/caaWorker/caaWorker.go

```diff
@@ -0,0 +1,164 @@
+		if err != nil {
+			result.Errors = append(result.Errors, fmt.Sprintf("CAA lookup failed for %s: %v", host, err))
+			continue
+		}
+
+		if res.Rcode != dns.RcodeSuccess {
+			result.Errors = append(result.Errors, fmt.Sprintf("CAA lookup failed for %s with %s", host, dns.RcodeToString[res.Rcode]))
+			continue
+		}
+
+		for _, rr := range res.Answer {
+			if caa, ok := rr.(*dns.CAA); ok {
+				caaRes.HasCAA = true
+				if caa.Tag == "issue" {
+					caaRes.IssueCAs = append(caaRes.IssueCAs, caa.Value)
+				} else if caa.Tag == "issuewild" {
+					caaRes.IssueWildCAs = append(caaRes.IssueWildCAs, caa.Value)
+				}
+			}
+		}
+
+		if caaRes.HasCAA {
+			caaRes.Host = host
+			break
+		}
+	}
+
+	res, err := json.Marshal(caaRes)
+	if err != nil {
+		result.Errors = append(result.Errors, err.Error())
+		result.Success = false
+	} else {
+		result.Success = true
+		result.Result = res
+	}
+
+	resChan <- result
+}
+
+// Assertor compares 2 caaResults and reports differences.
+func (e eval) Assertor(caaResult, assertresults []byte) (pass bool, body []byte, err error) {
+	var result, assertres Result
+	pass = false
+	err = json.Unmarshal(caaResult, &result)
+	if err != nil {
+		return
+	}
+	err = json.Unmarshal(assertresults, &assertres)
+	if err != nil {
+		return
+	}
+
+	if result.HasCAA != assertres.HasCAA {
+		body = []byte("CAA mismatch")
+		return
+	}
+	if result.Host != assertres.Host {
+		body = []byte(fmt.Sprintf(`Assertion failed MatchedHost= %s`,
+			result.Host))
+		return
+	}
+
+	if len(result.IssueCAs) != len(assertres.IssueCAs) {
+		body = []byte("Issue CAs count mismatch")
+		return
+	}
+
+	for i := range result.IssueCAs {
+		if result.IssueCAs[i] != assertres.IssueCAs[i] {
+			body = []byte(fmt.Sprintf(`Issue CAs mismatch %s != %s`,
+				result.IssueCAs[i], assertres.IssueCAs[i]))
+			return
+		}
+	}
+
+	if len(result.IssueWildCAs) != len(assertres.IssueWildCAs) {
+		body = []byte("Issue CAs count mismatch")
+		return
+	}
+
+	for i := range result.IssueWildCAs {
+		if result.IssueWildCAs[i] != assertres.IssueWildCAs[i] {
+			body = []byte(fmt.Sprintf(`Issue CAs mismatch %s != %s`,
+				result.IssueWildCAs[i], assertres.IssueWildCAs[i]))
+			return
+		}
+	}
+
+	pass = true
+	return
+}
+
+func (e eval) AnalysisPrinter(input []byte, printAll interface{}) (results []string, err error) {
+	var r Result
+	err = json.Unmarshal(input, &r)
+	if err != nil {
+		err = fmt.Errorf("CAA worker: failed to parse results: %v", err)
+		return
+	}
+
+	if !r.HasCAA {
+		results = append(results, "No avaliable CAA records")
+	} else {
+		results = append(results, fmt.Sprintf("Available CAA records for domain: %s", r.Host))
+		if len(r.IssueCAs) > 0 {
+			results = append(results, "- CAs permitted to issue certificates:")
+			for _, issue := range r.IssueCAs {
+				results = append(results, fmt.Sprintf("  * %s", issue))
+			}
+		}
+		if len(r.IssueWildCAs) > 0 {
+			results = append(results, "- CAs permitted to issue wildcard certificates:")
```

Comment:
Add 2 white spaces before the hyphen: `  - CA $CA permitted to issue wildcard certs for $host`